### PR TITLE
Added upgrade suite to upgrade from 4.x cdn to 4.x latest

### PR DESCRIPTION
**Pull request by vivekanandan-k-rh**
Added test suite for upgrade from 4.x GAed version to 4.x latest build

Logs: http://magna002.ceph.redhat.com/ceph-qe-logs/vivek/cephci-run-9DJT80/

### CODE REVIEW

**Comment by udaysk23** [comment]
@vivekanandan-k-rh Could you please initiate one run and attach logs

File: suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+          dashboard_enabled: False
+          upgrade_ceph_packages: True
+          ceph_rhcs_version: 4
+          ceph_iscsi_config_dev: false
+          ceph_conf_overrides:
+            global:
+              osd_pool_default_pg_num: 64
+              osd_default_pool_size: 2
+              osd_pool_default_pgp_num: 64
+              mon_max_pg_per_osd: 1024
```
Comment (udaysk23) [inline suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml:231]: Can we skip this part if not required

File: suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+          osd_auto_discovery: False
+          ceph_docker_registry_auth: false
+          dashboard_enabled: false
+          copy_admin_key: True
+          ceph_conf_overrides:
+            global:
+              osd_pool_default_pg_num: 128
+              osd_default_pool_size: 2
+              osd_pool_default_pgp_num: 128
+              mon_max_pg_per_osd: 4096
```
Comment (udaysk23) [inline suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml:30]: Can we skip this part if not required

File: suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+          osd_scenario: lvm
+          osd_auto_discovery: False
+          ceph_docker_registry_auth: false
+          dashboard_enabled: false
+          copy_admin_key: True
+          ceph_conf_overrides:
+            client:
+              rgw crypt require ssl: false
+              rgw crypt s3 kms encryption keys: testkey-1=YmluCmJvb3N0CmJvb3N0LWJ1aWxkCmNlcGguY29uZgo=
+                testkey-2=aWIKTWFrZWZpbGUKbWFuCm91dApzcmMKVGVzdGluZwo=
```
Comment (psathyan) [inline suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml:29]: can we modify these lines to
```
              rgw crypt s3 kms encryption keys: 
                testkey-1: YmluCmJvb3N0CmJvb3N0LWJ1aWxkCmNlcGguY29uZgo=
                testkey-2: aWIKTWFrZWZpbGUKbWFuCm91dApzcmMKVGVzdGluZwo=
```

File: suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+        timeout: 500
+  - test:
+      name: Resharding tests
+      desc: Reshading test - dynamic
+      polarion-id: CEPH-83571740
+      module: sanity_rgw.py
+      config:
+        script-name: test_dynamic_bucket_resharding.py
+        config-file-name: test_dynamic_resharding.yaml
+        timeout: 500
```
Comment (psathyan) [inline suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml:197]: Wouldn't these test fail with versions lower than 4.2?

**Review by psathyan — COMMENTED** [review]
IMO, we would minimize the number of tests we are running prior to the upgrade. Instead opt to generate objects (versioning) for 25% capacity and then perform the upgrade. Post the upgrade, we would run the intented tests.

File: suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml [hunk 5]
```diff
@@ -0,0 +1,383 @@
+tests:
```
[the remark this answers is not in the document]
Comment (vivekanandan-k-rh) [reply suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml]: Done

File: suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml (hunk 3 above)
Comment (vivekanandan-k-rh) [reply suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml]: Done

File: suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml (hunk 4 above)
Comment (vivekanandan-k-rh) [reply suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml]: @psathyan . Yes. it was mainly added for 4.2 and above

**Comment by vivekanandan-k-rh** [comment]
> @vivekanandan-k-rh Could you please initiate one run and attach logs

Done. http://magna002.ceph.redhat.com/ceph-qe-logs/vivek/cephci-run-9DJT80/

File: suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml (hunk 4 above)
Comment (psathyan) [reply suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml]: Can we rename the file name to `rgw_test-update-4.2-to-4.x-latest.yaml`?

Comment (vivekanandan-k-rh) [reply suites/nautilus/rgw/upgrade_rgw_singlesite_4.x_cdn_to_4.x_latest.yaml]: Done

**Review by psathyan — APPROVED** [review]

**Pull request merged**